sonic_platform_base/sonic_sfp/sfputilbase.py

```diff
@@ -369,6 +369,38 @@ def _read_eeprom_devid(self, port_num, devid, offset, num_bytes = 256):
 
         return eeprom_raw
 
+    def _write_eeprom_specific_bytes(self, sysfsfile_eeprom, offset, num_bytes, write_buffer):
+        try:
+            sysfsfile_eeprom.seek(offset)
+            sysfsfile_eeprom.write(write_buffer)
+            sysfsfile_eeprom.close()
```

Comment:
I would suggest not closing a file descriptor provided by a parent function.
The `_write_eeprom_devid` which is currently the only user is already taking care of this.

If what you want to flush the buffer you can call `sysfsfile_eeprom.flush()`.
However adding `buffering=0` to `open` would prevent any buffering from happening in the first place, which I recommend in this case.

Here is an example that would be problematic by keeping the close here.
```
with open(path, mode='wb', buffering=0) as eeprom:
    self._write_eeprom_specific_bytes(eeprom, 0, 3, b'abc')
    self._write_eeprom_specific_bytes(eeprom, 3, 3, b'def') # exception because eeprom is closed
```

Reply:
fixed